p2p/src/peer_manager/dns_seed.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2021-2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use common::chain::{config::ChainType, ChainConfig};
+use crypto::random::{make_pseudo_rng, seq::IteratorRandom};
+use logging::log;
+use p2p_types::socket_address::SocketAddress;
+
+use crate::config::P2pConfig;
+
+#[async_trait]
+pub trait DnsSeed: Send + Sync {
+    async fn obtain_addresses(&self) -> Vec<SocketAddress>;
+}
+
+pub struct DefaultDnsSeed {
+    chain_config: Arc<ChainConfig>,
+    p2p_config: Arc<P2pConfig>,
+}
+
+impl DefaultDnsSeed {
+    pub fn new(chain_config: Arc<ChainConfig>, p2p_config: Arc<P2pConfig>) -> Self {
+        Self {
+            chain_config,
+            p2p_config,
+        }
+    }
+}
+
+/// Hardcoded seed DNS hostnames
+// TODO: Replace with actual values
+const DNS_SEEDS_MAINNET: [&str; 0] = [];
+const DNS_SEEDS_TESTNET: [&str; 1] = ["testnet-seed.mintlayer.org"];
+
+/// Maximum number of records accepted in a single DNS server response
+const MAX_DNS_RECORDS: usize = 10;
+
+#[async_trait]
+impl DnsSeed for DefaultDnsSeed {
+    async fn obtain_addresses(&self) -> Vec<SocketAddress> {
+        let dns_seed = match self.chain_config.chain_type() {
```

Comment:
FYI: the logic in this function is not new; it was moved here from `reload_dns_seed`